stable_baselines3/her/her.py

```diff
@@ -0,0 +1,416 @@
+        gamma: float = 0.99,
+        train_freq: int = 1,
+        gradient_steps: int = 1,
+        n_episodes_rollout: int = -1,
+        action_noise: Optional[ActionNoise] = None,
+        optimize_memory_usage: bool = False,
+        policy_kwargs: Dict[str, Any] = None,
+        tensorboard_log: Optional[str] = None,
+        verbose: int = 0,
+        device: Union[th.device, str] = "cpu",
+        support_multi_env: bool = False,
+        create_eval_env: bool = False,
+        monitor_wrapper: bool = True,
+        seed: Optional[int] = None,
+        use_sde: bool = False,
+        sde_sample_freq: int = -1,
+        use_sde_at_warmup: bool = False,
+        sde_support: bool = True,
+        *args,
+        **kwargs
+    ):
+
+        if isinstance(goal_strategy, str):
+            self.goal_strategy = KEY_TO_GOAL_STRATEGY[goal_strategy.lower()]
+        else:
+            self.goal_strategy = goal_strategy
+
+        assert isinstance(
+            self.goal_strategy, GoalSelectionStrategy
+        ), "Invalid goal selection strategy," "please use one of {}".format(list(GoalSelectionStrategy))
+
+        self.env = ObsWrapper(env)
+
+        # get arguments for the model initialization
+        model_signature = signature(model.__init__)
+        arguments = locals()
+        model_init_dict = {
+            key: arguments[key]
+            for key in model_signature.parameters.keys()
+            if key in arguments and key != "self" and key != "env"
+        }
+
+        super(HER, self).__init__(
+            policy,
+            self.env,
+            BasePolicy,
+            learning_rate,
+            buffer_size,
+            learning_starts,
+            batch_size,
+            tau,
+            gamma,
+            train_freq,
+            gradient_steps,
+            n_episodes_rollout,
+            action_noise,
+            optimize_memory_usage,
+            policy_kwargs,
+            tensorboard_log,
+            verbose,
+            device,
+            support_multi_env,
+            create_eval_env,
+            monitor_wrapper,
+            seed,
+            use_sde,
+            sde_sample_freq,
+            use_sde_at_warmup,
+            sde_support,
+        )
+
+        # model initialization
+        self.model = model(env=self.env, **model_init_dict, **kwargs)
+
+        # if we sample her transitions online use custom replay buffer
+        self.online_sampling = online_sampling
+        if self.online_sampling:
+            self.model.replay_buffer = HerReplayBuffer(
+                self.env,
+                buffer_size,
+                self.goal_strategy,
+                self.env.observation_space,
+                self.env.action_space,
+                device,
+                self.n_envs,
+                her_ratio,
+            )
+
+        # storage for transitions of current episode
+        self.episode_storage = []
+        self.n_goals = n_goals
+
+    def learn(
```

Comment:
Not exactly. Normally we would like to use the learn method of the model class but since I changed the collect_rollout method we have to specify that we use the new one, right?

Reply:
I don't think so. Should work if you overwrite the `collect_rollout` only.